swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -399,6 +399,29 @@ private module ReturnNodes {
     override ReturnKind getKind() { result instanceof NormalReturnKind }
   }
 
+  class SelfReturnNode extends ReturnNode, TInoutReturnNode, NodeImpl {
```

Comment:
Do we really need to create a whole new class for this? Can we not just inherit from `InoutReturnNodeImpl`, restrict the `exit` member already present in that class and override `getKind`?

Reply:
This has been fixed in https://github.com/github/codeql/pull/12415/commits/86fd21de452bb83850fea9182b3bbd2d78876e7a.